### PR TITLE
40 remove autofac

### PR DESCRIPTION
**Pull request by Metawolve**
Removed autofac, refactored the startup to use the default DI pipeline and the minimal hosting model introduced in .NET 6 with extension methods.

### CODE REVIEW

File: src/Eurofurence.App.Server.Services/Abstractions/JobOption.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Eurofurence.App.Server.Services.Abstractions
+{
+    public class JobOption
+    {
+        public bool Enabled { get; set; }
+
```
Comment (maakinoh) [inline src/Eurofurence.App.Server.Services/Abstractions/JobOption.cs:6]: Extra line here

**Review by maakinoh — APPROVED** [review]
Looks good to me.

Just some minor issues

**Comment by maakinoh** [comment]
Not really an issue due to #250. But the Program.cs file of the CLI module is still importing Autofac

File: src/Eurofurence.App.Server.Services/Abstractions/ArtistsAlley/ArtistAlleyOptions.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Eurofurence.App.Server.Services.Abstractions.ArtistsAlley
+{
+    public class ArtistAlleyOptions
+    {
+        public bool RegistrationEnabled { get; set; }
+        public ArtistAlleyTelegramOptions Telegram { get; set; }
```
[the remark this answers is not in the document]
Comment (Metawolve) [reply src/Eurofurence.App.Server.Services/Abstractions/ArtistsAlley/ArtistAlleyOptions.cs]: Sounds good, changed them to init-only and checked that they still work^^

File: src/Eurofurence.App.Server.Services/Abstractions/JobOption.cs (hunk 1 above)
Comment (Metawolve) [reply src/Eurofurence.App.Server.Services/Abstractions/JobOption.cs]: Removed

**Comment by Metawolve** [comment]
Autofac is still imported in the CLItools and they currently don't work without it so they don't compile after this PR. But due to #250 I think we don't need to put in the time to make that work.

**Pull request merged**